dart.go

```diff
@@ -0,0 +1,756 @@
+package colfer
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"text/template"
+
+	"github.com/pascaldekloe/name"
+)
+
+// dartKeywords are the reserved tokens for Dart code.
+// Some entries are redundant due to the use of a Go parser.
+var dartKeywords = map[string]struct{}{
+	"abstract": {}, "as": {}, "assert": {}, "async": {}, "await": {},
+	"break": {}, "case": {}, "catch": {}, "class": {}, "const": {},
+	"continue": {}, "covariant": {}, "default": {}, "deferred": {},
+	"do": {}, "dynamic": {}, "else": {}, "enum": {}, "export": {},
+	"extends": {}, "extension": {}, "external": {}, "factory": {},
+	"false": {}, "final": {}, "finally": {}, "for": {}, "Function": {},
+	"get": {}, "hide": {}, "if": {}, "implements": {}, "import": {},
+	"in": {}, "interface": {}, "is": {}, "library": {}, "mixin": {},
+	"new": {}, "null": {}, "on": {}, "operator": {}, "part": {},
+	"rethrow": {}, "return": {}, "set": {}, "show": {}, "static": {},
+	"super": {}, "switch": {}, "sync": {}, "this": {}, "throw": {},
+	"true": {}, "try": {}, "typedef": {}, "var": {}, "void": {},
+	"while": {}, "with": {}, "yield": {},
+	// and some more not to interfere with properties
```

Comment:
Ouch! We can test this with an extra entry in `testdata/break.colf`. Add a variable somewhere and you'll have a bug when it is not explicitly listed here. Is there really no way to generate more robust code which allows overlap with property names?

Reply:
I can't think of a more robust solution than this. I mean if we go with `__a__`, `__buf__` there's always a possibility that the user adds a variable like that. Except we deny it for all languages. What are your thoughts about this?

Reply:
You may have an idea there. Colfer does not allow leading underscores. Leading underscores are also the traditional way to distinguish "system" names. I like it. 🙂

Reply:
I just checked underscore prefix in Colfer schema, and it passed. But it's actually a good idea, should be added I guess.

Reply:
Another one related here: I checked for lint errors, and got one with `as_` for

```// As tests binary lists.
	as []binary
```

 in `gen`. It says `Name non-constant identifiers using lowerCamelCase.` If I rename it to `as_0` it disappears, but that may overlap with another property of the same object, if underscore is only forbidden in front. What do you think, should we address this?

Reply:
Sorry, my bad, underscore was actually removed in `{{.NameNative}}`, it stayed in `{{.String}}` only, but that doesn't hurt I think.

Reply:
Misread your last comment as if the issue was resolved. So there is this linter thing, and you just patched it with the `_0` solution. No idea what the problem is we're fixing. 😄 I think it is a good idea to add the linter to the test target in Makefile.

Reply:
Ok, I'll try to find it. So far I dropped the output in vs code, and it gave me the lint warnings :)